backend/src/test/java/ddangkong/service/balance/room/RoomServiceTest.java

```diff
@@ -0,0 +1,68 @@
+package ddangkong.service.balance.room;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import ddangkong.controller.balance.member.dto.MemberResponse;
+import ddangkong.controller.balance.room.dto.RoomJoinResponse;
+import ddangkong.exception.BadRequestException;
+import ddangkong.service.BaseServiceTest;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class RoomServiceTest extends BaseServiceTest {
+
+    @Autowired
+    private RoomService roomService;
+
+    @Nested
+    class 방_생성 {
+
+        @Test
+        void 방_생성_시_멤버를_생성하고_방을_생성한다() {
+            // given
+            String nickname = "나는방장";
+            MemberResponse expectedMemberResponse = new MemberResponse(5L, nickname, true);
+            RoomJoinResponse expected = new RoomJoinResponse(3L, expectedMemberResponse);
+
+            // when
+            RoomJoinResponse roomServiceRoom = roomService.createRoom(nickname);
+
+            // then
+            assertThat(roomServiceRoom).isEqualTo(expected);
+        }
+    }
+
+    @Nested
+    class 방_참여 {
+
+        @Test
+        void 이미_생성된_방에_참여한다() {
+            // given
+            String nickname = "나는참가자";
+            Long joinRoomId = 2L;
+            MemberResponse expectedMemberResponse = new MemberResponse(5L, nickname, false);
+            RoomJoinResponse expected = new RoomJoinResponse(joinRoomId, expectedMemberResponse);
+
+            // when
+            RoomJoinResponse roomServiceRoom = roomService.joinRoom(nickname, joinRoomId);
+
+            // then
+            assertThat(roomServiceRoom).isEqualTo(expected);
+        }
+
+        @Test
+        void 존재하지_않는_방에_참여시_예외를_던진다() {
+            // given & when
+            String nickname = "나는참가자";
+            Long nonExistId = 99999999999L;
+            MemberResponse expectedMemberResponse = new MemberResponse(5L, nickname, false);
+            RoomJoinResponse expected = new RoomJoinResponse(nonExistId, expectedMemberResponse);
+
+            // then
+            assertThatThrownBy(() -> roomService.joinRoom(nickname, nonExistId))
+                    .isInstanceOf(BadRequestException.class);
+        }
```

Comment:
의견) 사용하지 않는 변수들을 제거해 보았습니다~

```suggestion
        @Test
        void 존재하지_않는_방에_참여시_예외를_던진다() {
            // given
            String nickname = "나는참가자";
            Long nonExistId = 99999999999L;

            // when & then
            assertThatThrownBy(() -> roomService.joinRoom(nickname, nonExistId))
                    .isInstanceOf(BadRequestException.class);
        }
```